### ISSUE TITLE
Document `base` more clearly

### ISSUE THREAD

**Issue by Jutanium** [withastro/docs#1643]
Our current docs for [base](https://docs.astro.build/en/reference/configuration-reference/#base) says that we will start your server at the specified path (so `localhost:3000/base` rather than `localhost:3000`). But what else does it do? It can seem like it's supposed to affect your import paths (see [this support thread message](https://discord.com/channels/830184174198718474/1023967842112118895/1024674905251393546)), like the [`<base>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base) tag, but it doesn't.

**Comment by zihan-ch** [comment]
That is right. Should we open a new PR now?

**Comment by zihan-ch** [comment]
I am doing this at #1671.

**Comment by Jutanium** [comment]
[This](https://github.com/withastro/astro/pull/4992) documents the current behavior, but we now know that this current behavior is unintended. Waiting on [this issue](https://github.com/withastro/astro/issues/4988) to be fixed, and then we'll need to document the corrected behavior

**Comment by sarah11918** [comment]
Followup: No longer blocked, and https://github.com/withastro/astro/pull/5693 has been made to address this. Comments can be made there! 

Thanks for your contributions, everyone!
